networks/0_process_array_data.py

```diff
@@ -0,0 +1,101 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:core_acc] *
+#     language: python
+#     name: conda-env-core_acc-py
+# ---
+
+# # Process array data
+#
+# This notebook will process array data to:
+# 1. Include only those PAO1 samples, based on the metadata
+# 2. Use only genes shared by both the array and RNA-seq compendia
+
+# %load_ext autoreload
+# %autoreload 2
+import pandas as pd
+from core_acc_modules import paths
+
+# +
+# Files
+array_compendium_filename = paths.ARRAY_DATA
+array_metadata_filename = paths.ARRAY_METADATA
+
+pao1_rnaseq_compendium_filename = paths.PAO1_COMPENDIUM
+# -
+
+array_compendium = pd.read_csv(
+    array_compendium_filename, sep="\t", index_col=0, header=0
+).T
+array_metadata = pd.read_csv(array_metadata_filename, sep="\t", index_col=0, header=0)
+
+array_compendium.head()
+
+print(array_metadata.shape)
+array_metadata.head()
+
+pao1_strain_values = [
+    strain_name
+    for strain_name in array_metadata["strain"].unique()
+    if ("PAO1" in strain_name) & ("MPAO1" not in strain_name)
+]
+
+pao1_sample_ids = array_metadata[
+    array_metadata["strain"].apply(
+        lambda strain_name: strain_name in pao1_strain_values
+    )
+]["ml_data_source"]
+
+# Drop any sample ids that are na
+pao1_sample_ids.dropna(inplace=True)
```

Comment:
I _think_ the reason might be just in case you lose track of changes you've made to a dataframe that you've over-written. I usually just set a new variable, but it depends on what I'm doing.